### PR TITLE
Koa 5798 Clean up unnecessary variant snapshot generation

### PR DESCRIPTION
**Pull request by bvitaliyg**
+ [ ] Check this if you have read and followed the [contributing guidelines](https://github.com/Skyscanner/backpack-android/blob/main/CONTRIBUTING.md)

Remember to include the following changes:
+ [ ] Component `README.md`
+ [ ] Tests
+ [ ] Docs (either update [backpack-docs](https://github.com/Skyscanner/backpack-docs) now, or create a follow up ticket)

If you are curious about how we review, please read through the [code review guidelines](https://github.com/Skyscanner/backpack/blob/main/CODE_REVIEW_GUIDELINES.md)

### CODE REVIEW

File: app/src/androidTest/java/net/skyscanner/backpack/snackbar/BpkSnackbarTests.kt [hunk 1]
```diff
@@ -41,14 +43,16 @@ class BpkSnackbarTests : BpkSnapshotTest() {
   }
 
   @Test
-  fun defaultWithAction() {
+  @Variants(BpkTestVariant.Default)
```
Comment (marianeum) [inline app/src/androidTest/java/net/skyscanner/backpack/snackbar/BpkSnackbarTests.kt:46]: that has layout + colour implications, so all 4?

File: app/src/androidTest/java/net/skyscanner/backpack/snackbar/BpkSnackbarTests.kt [hunk 2]
```diff
@@ -58,6 +62,7 @@ class BpkSnackbarTests : BpkSnapshotTest() {
   }
 
   @Test
+  @Variants(BpkTestVariant.Default)
```
Comment (marianeum) [inline app/src/androidTest/java/net/skyscanner/backpack/snackbar/BpkSnackbarTests.kt:65]: same (for most of these tests)

File: app/src/androidTest/java/net/skyscanner/backpack/snackbar/BpkSnackbarTests.kt [hunk 3]
```diff
@@ -67,6 +72,7 @@ class BpkSnackbarTests : BpkSnapshotTest() {
   }
 
   @Test
+  @Variants(BpkTestVariant.Default, BpkTestVariant.DarkMode, BpkTestVariant.Rtl, BpkTestVariant.Themed)
```
Comment (marianeum) [inline app/src/androidTest/java/net/skyscanner/backpack/snackbar/BpkSnackbarTests.kt:75]: this is all 4, so no declaration needed

**Review by marianeum — COMMENTED** [review]
take a bunch of these as a suggestion. IMO it's better to run more tests unnecessarily as default will run all tests anyways than filtering out too many and miss bugs in cases where we made an assumption that something wouldn't change when it does

File: app/src/androidTest/java/net/skyscanner/backpack/barchart/BpkBarchartTests.kt [hunk 4]
```diff
@@ -51,7 +53,8 @@ class BpkBarchartTests : BpkSnapshotTest() {
   }
 
   @Test
-  fun half() {
+  @Variants(BpkTestVariant.Default)
```
[the remark this answers is not in the document]
Comment (bvitaliyg) [reply app/src/androidTest/java/net/skyscanner/backpack/barchart/BpkBarchartTests.kt:56]: We test DM in full variant. Themed is added

File: app/src/androidTest/java/net/skyscanner/backpack/calendar2/BpkCalendarTest.kt [hunk 5]
```diff
@@ -41,20 +43,23 @@ class BpkCalendarTest : BpkSnapshotTest() {
   }
 
   @Test
-  fun labeledCalendarDefault() {
+  @Variants(BpkTestVariant.Default, BpkTestVariant.DarkMode)
```
[the remark this answers is not in the document]
Comment (bvitaliyg) [reply app/src/androidTest/java/net/skyscanner/backpack/calendar2/BpkCalendarTest.kt:46]: The label is always centered, so it's the same in RTL. But the colors might be different. We also don't have theming properties for this case, so labels will always have the same color.

File: app/src/androidTest/java/net/skyscanner/backpack/calendar2/BpkCalendarTest.kt [hunk 6]
```diff
@@ -41,20 +43,23 @@ class BpkCalendarTest : BpkSnapshotTest() {
...
   fun past() {
     val calendar = BpkCalendar(testContext)
     calendar.setParams(BpkCalendarTestCases.Params.Past)
     snap(calendar, padding = 0)
   }
 
   @Test
+  @Variants(BpkTestVariant.Default)
   fun leapFebruary() {
```
[the remark this answers is not in the document]
Comment (bvitaliyg) [reply app/src/androidTest/java/net/skyscanner/backpack/calendar2/BpkCalendarTest.kt:63]: Not all – this one checks the visual layout. But behavioral tests should, yes

File: app/src/androidTest/java/net/skyscanner/backpack/calendar2/BpkCalendarTest.kt [hunk 7]
```diff
@@ -103,6 +113,7 @@ class BpkCalendarTest : BpkSnapshotTest() {
   }
 
   @Test
+  @Variants(BpkTestVariant.Default, BpkTestVariant.DarkMode)
```
[the remark this answers is not in the document]
Comment (bvitaliyg) [reply app/src/androidTest/java/net/skyscanner/backpack/calendar2/BpkCalendarTest.kt:116]: We only have theming for selection for Calendar

File: app/src/androidTest/java/net/skyscanner/backpack/calendar2/BpkCalendarTest.kt [hunk 8]
```diff
@@ -126,6 +138,7 @@ class BpkCalendarTest : BpkSnapshotTest() {
   }
 
   @Test
+  @Variants(BpkTestVariant.Default)
```
[the remark this answers is not in the document]
Comment (bvitaliyg) [reply app/src/androidTest/java/net/skyscanner/backpack/calendar2/BpkCalendarTest.kt:141]: Yes, those need to be instrumentation tests

File: app/src/androidTest/java/net/skyscanner/backpack/compose/floatingnotification/BpkFloatingNotificationTest.kt [hunk 9]
```diff
@@ -48,6 +50,7 @@ class BpkFloatingNotificationTest : BpkSnapshotTest() {
   }
 
   @Test
+  @Variants(BpkTestVariant.Default)
```
[the remark this answers is not in the document]
Comment (bvitaliyg) [reply app/src/androidTest/java/net/skyscanner/backpack/compose/floatingnotification/BpkFloatingNotificationTest.kt:53]: We test all these variants in withIconAndCta – it's been tested for all of the cases

File: app/src/androidTest/java/net/skyscanner/backpack/compose/floatingnotification/BpkFloatingNotificationTest.kt [hunk 10]
```diff
@@ -61,6 +64,7 @@ class BpkFloatingNotificationTest : BpkSnapshotTest() {
...
```
[the remark this answers is not in the document]
Comment (bvitaliyg) [reply app/src/androidTest/java/net/skyscanner/backpack/compose/floatingnotification/BpkFloatingNotificationTest.kt:67]: Same, let's keep the discussion in one place

File: app/src/androidTest/java/net/skyscanner/backpack/text/BpkTextInputLayoutTest.kt [hunk 11]
```diff
@@ -42,31 +44,36 @@ class BpkTextInputLayoutTest : BpkSnapshotTest() {
...
 
   @Test
+  @Variants(BpkTestVariant.Default)
   fun error() {
     subject.error = "Error"
     snap(subject)
   }
 
   @Test
+  @Variants(BpkTestVariant.Default)
```
[the remark this answers is not in the document]
Comment (bvitaliyg) [reply app/src/androidTest/java/net/skyscanner/backpack/text/BpkTextInputLayoutTest.kt:61]: We test it in helperAndError

File: app/src/androidTest/java/net/skyscanner/backpack/compose/textfield/BpkTextFieldTest.kt [hunk 12]
```diff
@@ -70,6 +70,7 @@ class BpkTextFieldTest : BpkSnapshotTest() {
   }
 
   @Test
+  @Variants(BpkTestVariant.Default, BpkTestVariant.Rtl)
```
[the remark this answers is not in the document]
Comment (bvitaliyg) [reply app/src/androidTest/java/net/skyscanner/backpack/compose/textfield/BpkTextFieldTest.kt:73]: It is RTL :)

File: app/src/androidTest/java/net/skyscanner/backpack/compose/textfield/BpkTextFieldTest.kt [hunk 13]
```diff
@@ -125,6 +126,7 @@ class BpkTextFieldTest : BpkSnapshotTest() {
   }
 
   @Test
+  @Variants(BpkTestVariant.Default)
```
[the remark this answers is not in the document]
Comment (bvitaliyg) [reply app/src/androidTest/java/net/skyscanner/backpack/compose/textfield/BpkTextFieldTest.kt:129]: Not sure about this one as it only shown text align of multiline text, but I guess it won't harm

File: app/src/androidTest/java/net/skyscanner/backpack/icon/BpkIconTest.kt [hunk 14]
```diff
@@ -35,15 +35,17 @@ import org.junit.Test
 class BpkIconTest : BpkSnapshotTest() {
 
   @Test
-  fun allSmallIcons() {
+  @Variants(BpkTestVariant.Default, BpkTestVariant.DarkMode, BpkTestVariant.Rtl)
```
[the remark this answers is not in the document]
Comment (bvitaliyg) [reply app/src/androidTest/java/net/skyscanner/backpack/icon/BpkIconTest.kt:38]: What about auto-mirrored icons?

File: app/src/androidTest/java/net/skyscanner/backpack/map/internal/BpkMapMarkerTest.kt [hunk 15]
```diff
@@ -42,32 +44,37 @@ class BpkMapMarkerTest : BpkSnapshotTest() {
   }
 
   @Test
+  @Variants(BpkTestVariant.Default, BpkTestVariant.DarkMode)
```
[the remark this answers is not in the document]
Comment (bvitaliyg) [reply app/src/androidTest/java/net/skyscanner/backpack/map/internal/BpkMapMarkerTest.kt:47]: We don't have theming in MapMarkers

File: app/src/androidTest/java/net/skyscanner/backpack/map/internal/BpkMapMarkerTest.kt [hunk 16]
```diff
@@ -42,32 +44,37 @@ class BpkMapMarkerTest : BpkSnapshotTest() {
...
 
   @Test
+  @Variants(BpkTestVariant.Default)
   fun noPointer() {
     val view = createMarker(0, false)
     snap(view)
   }
 
   @Test
+  @Variants(BpkTestVariant.Default, BpkTestVariant.DarkMode, BpkTestVariant.Rtl)
```
[the remark this answers is not in the document]
Comment (bvitaliyg) [reply app/src/androidTest/java/net/skyscanner/backpack/map/internal/BpkMapMarkerTest.kt:62]: Same

File: app/src/androidTest/java/net/skyscanner/backpack/navbar/BpkNavBarTest.kt [hunk 17]
```diff
@@ -47,6 +49,7 @@ class BpkNavBarTest : BpkSnapshotTest() {
   }
 
   @Test
+  @Variants(BpkTestVariant.Default)
```
[the remark this answers is not in the document]
Comment (bvitaliyg) [reply app/src/androidTest/java/net/skyscanner/backpack/navbar/BpkNavBarTest.kt:52]: That's why we test everything when it's all combined, e.g. in collapsed_iconAndMenu

File: app/src/androidTest/java/net/skyscanner/backpack/nudger/BpkNudgerTest.kt [hunk 18]
```diff
@@ -35,20 +37,24 @@ class BpkNudgerTest : BpkSnapshotTest() {
   }
 
   @Test
+  fun default() {
+    nudger.value = 5
+    snap(nudger)
+  }
+
+  @Test
+  @Variants(BpkTestVariant.Default)
```
[the remark this answers is not in the document]
Comment (bvitaliyg) [reply app/src/androidTest/java/net/skyscanner/backpack/nudger/BpkNudgerTest.kt:46]: True. Although in this case we test BpkButton and not BpkNudger (in the dark mode).

File: app/src/androidTest/java/net/skyscanner/backpack/overlay/BpkOverlayTest.kt [hunk 19]
```diff
@@ -67,20 +69,21 @@ class BpkOverlayTest : BpkSnapshotTest() {
   }
 
   @Test
-  @Ignore("Screenshot testing library does not capture rounded corners")
```
[the remark this answers is not in the document]
Comment (bvitaliyg) [reply app/src/androidTest/java/net/skyscanner/backpack/overlay/BpkOverlayTest.kt:70]: It is. I just needed to test it for Shot

File: app/src/androidTest/java/net/skyscanner/backpack/rating/BpkRatingIconsTest.kt [hunk 20]
```diff
@@ -20,25 +20,30 @@ package net.skyscanner.backpack.rating
...
+import net.skyscanner.backpack.BpkTestVariant
+import net.skyscanner.backpack.Variants
 import org.junit.Test
 import org.junit.runner.RunWith
 
 @RunWith(AndroidJUnit4::class)
 class BpkRatingIconsTest : BpkSnapshotTest() {
 
   @Test
+  @Variants(BpkTestVariant.Default)
```
[the remark this answers is not in the document]
Comment (bvitaliyg) [reply app/src/androidTest/java/net/skyscanner/backpack/rating/BpkRatingIconsTest.kt:32]: It only tests that for a given value there's a correct icon set. It's a behavioral test. Icons are tested in vertical/horizontal cases with dark mode, but not with the theme, as the icon color is not a themable attribute.

File: app/src/androidTest/java/net/skyscanner/backpack/rating/BpkRatingValuesTest.kt [hunk 21]
```diff
@@ -20,61 +20,72 @@ package net.skyscanner.backpack.rating
...
+import net.skyscanner.backpack.BpkTestVariant
+import net.skyscanner.backpack.Variants
 import org.junit.Test
 import org.junit.runner.RunWith
 
 @RunWith(AndroidJUnit4::class)
 class BpkRatingValuesTest : BpkSnapshotTest() {
 
   @Test
+  @Variants(BpkTestVariant.Default)
```
[the remark this answers is not in the document]
Comment (bvitaliyg) [reply app/src/androidTest/java/net/skyscanner/backpack/rating/BpkRatingValuesTest.kt:32]: Same, it just testes text being set.

File: app/src/androidTest/java/net/skyscanner/backpack/snackbar/BpkSnackbarTests.kt (hunk 1 above)
Comment (bvitaliyg) [reply app/src/androidTest/java/net/skyscanner/backpack/snackbar/BpkSnackbarTests.kt:46]: We test it all in titleWithIconAndAction

File: app/src/androidTest/java/net/skyscanner/backpack/snackbar/BpkSnackbarTests.kt (hunk 2 above)
Comment (bvitaliyg) [reply app/src/androidTest/java/net/skyscanner/backpack/snackbar/BpkSnackbarTests.kt:65]: Let's keep the thread in one place

File: app/src/androidTest/java/net/skyscanner/backpack/text/BpkTextFieldTests.kt [hunk 22]
```diff
@@ -41,37 +43,43 @@ class BpkTextFieldTests : BpkSnapshotTest() {
...
 
   @Test
+  @Variants(BpkTestVariant.Default, BpkTestVariant.DarkMode)
   fun disabled() {
     subject.isEnabled = false
     snap(subject)
   }
 
   @Test
+  @Variants(BpkTestVariant.Default, BpkTestVariant.Rtl)
```
[the remark this answers is not in the document]
Comment (bvitaliyg) [reply app/src/androidTest/java/net/skyscanner/backpack/text/BpkTextFieldTests.kt:60]: Colors are being tested in iconStart_withTint

File: app/src/androidTest/java/net/skyscanner/backpack/text/BpkTextFieldTests.kt [hunk 23]
```diff
@@ -41,37 +43,43 @@ class BpkTextFieldTests : BpkSnapshotTest() {
...
   fun iconEnd() {
     subject.iconEnd = AppCompatResources.getDrawable(testContext, R.drawable.bpk_search)
     snap(subject)
   }
 
   @Test
+  @Variants(BpkTestVariant.Default, BpkTestVariant.DarkMode)
   fun iconStart_withTint() {
```
[the remark this answers is not in the document]
Comment (bvitaliyg) [reply app/src/androidTest/java/net/skyscanner/backpack/text/BpkTextFieldTests.kt:75]: I guess (from git history) it just needs to make sure the icons have the same colors as the text. But I don't think e need it anymore, so I'm removing it.

File: app/src/androidTest/java/net/skyscanner/backpack/text/BpkTextInputLayoutTest.kt [hunk 24]
```diff
@@ -42,31 +44,36 @@ class BpkTextInputLayoutTest : BpkSnapshotTest() {
...
 
   @Test
+  @Variants(BpkTestVariant.Default, BpkTestVariant.DarkMode)
   fun disabled() {
     textField.isEnabled = false
     snap(subject)
   }
 
   @Test
+  @Variants(BpkTestVariant.Default)
```
[the remark this answers is not in the document]
Comment (bvitaliyg) [reply app/src/androidTest/java/net/skyscanner/backpack/text/BpkTextInputLayoutTest.kt:54]: It's being tested in helperAndError

**Review by marianeum — APPROVED** [review]

**Pull request merged**